core/src/test/java/io/grpc/DeadlineTest.java

```diff
@@ -69,15 +71,19 @@ public void shortDeadlineEventuallyExpires() throws Exception {
     assertTrue(deadline.timeRemaining(TimeUnit.NANOSECONDS) > 0);
     assertFalse(deadline.isExpired());
     Thread.sleep(101);
+
     assertTrue(deadline.isExpired());
-    assertEquals(0, deadline.timeRemaining(TimeUnit.NANOSECONDS), delta);
+    long actualDelta = deadline.timeRemaining(NANOSECONDS);
+    assertTrue(maxDelta + " < " + actualDelta, maxDelta >= actualDelta);
```

Comment:
This is exactly what the original [assertEquals(double, double, double)](http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(double, double, double)) is doing. I don't believe it would improvement the flakiness.

The reason of #1617 is that the actual delta is ~6E8 ns == ~600 ms > 20 ms (the delta). We can fix it by tuning up `maxDelta` to a larger value.